src/hooks/protocol/use-configure-campaign.ts

```diff
@@ -8,12 +8,15 @@ import { ERRORS } from '@notifications/errors.ts';
 import { useAccountSession } from '@src/hooks/use-account-session.ts';
 import { ConfigureCampaignParams, UseConfigureCampaignHook } from '@src/hooks/protocol/types.ts';
 import { notifyError } from '@notifications/internal-notifications.ts';
+import {useSelector} from "react-redux"
+import {RootState} from "@redux/store.ts"
 
 export const useConfigureCampaign = (): UseConfigureCampaignHook => {
   const [data, setData] = useState<any>(null);
   const [loading, setLoading] = useState<boolean>(false);
   const { bundlerClient, smartAccount } = useWeb3Session();
-  const { isAuthenticated, logout } = useAccountSession();
+  const { logout } = useAccountSession();
+  const isAuthenticated = useSelector((state: RootState) => state.auth.isFullyAuthenticated);
```

Comment:
Se repite por todo el código esta lógica, tiene que existir una manera de que la arquitectura que estés diseñando te permita controlar todo esto de manera simplificada, unificada y centralizada, no deberían existir lógicas separadas para manejar la sesión, y consultar el estado de la sesión

Reply:
Después de leer esto, https://www.syncfusion.com/blogs/post/react-hooks-vs-redux podrías ver si esta es la mejor opcion

Reply:
https://prateeksha.com/blog/react-hooks-vs-redux-in-2025-which-state-management-solution-is-better